### DISCUSSION TITLE
SLANG Support for rendering and shader script

### DISCUSSION THREAD

**Discussion by m21-cerutti** [godotengine/godot-proposals#11310] [Engine Core]
Hello,
I have seen recently about SLANG language for writing shaders.
https://www.khronos.org/news/press/khronos-group-launches-slang-initiative-hosting-open-source-compiler-contributed-by-nvidia
But I don't know if it planned in Godot and there is no proposition or discussion about it. I think it's promising and since it would allow multiple backend support natively, I think it could be a good solution ? The current custom GLSL language was for maintaining compatibility between different support from what I read, so maybe interesting to anticipate what it seems would be a new standard ?
Just opening the discussion.

**Comment by m21-cerutti** [comment]
There is also discussions in
https://godotforums.org/d/41560-khronos-group-launches-slang-initiative
https://www.reddit.com/r/godot/comments/1cob0ks/has_anyone_tried_slang_to_write_shaders_for_godot/

**Comment by AThousandShips** [comment]
I'd say we should consider using it *when* it has become universally supported on all relevant hardware and drivers

Also I'm not sure it can be expected to work on lower end contexts such as gles3 so we'd have to maintain two different comtexts and languages, which I don't know is a good idea, unless of course it'll be supported on webgl etc.

**Comment by Calinou** [comment]
Slang is mostly used in scientific circles right now, as opposed to game development. It's also notoriously slow at compiling shaders, which is problematic when you're iterating quickly on a shader and hoping the real-time preview will update as quickly as possible.
